src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -161,6 +162,17 @@ private void SetRedirection()
 
                 VerboseTrace(_handler._maxAutomaticRedirections.ToString());
                 SetCurlOption(CURLoption.CURLOPT_FOLLOWLOCATION, 1L);
+                if (string.Equals(_requestMessage.RequestUri.Scheme, UriSchemeHttps, StringComparison.OrdinalIgnoreCase))
+                {
+                    // if request scheme is https, follow only https
+                    SetCurlOption(CURLoption.CURLOPT_REDIR_PROTOCOLS, CurlProtocols.CURLPROTO_HTTPS);
```

Comment:
Kindly note that this change here does not fix  #3555 completely.  [GetAsync_AllowAutoRedirectTrue_RedirectFromHttpsToHttp_StatusCodeRedirect](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs#L209) will continue to fail. CurlHandler throws the exception (while winhttpHandler returns a Redirect):

        System.Net.Http.HttpRequestException : An error occurred while sending the request.
        ---- System.Net.Http.CurlException : Unsupported protocol
        Stack Trace:
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
            at System.Net.Http.Functional.Tests.HttpClientHandlerTest.

But I think this is the right thing to do. I will try and fix #3555 in a separate PR. Please let me know if am wrong.